server/src/domain/library/library.service.ts

```diff
@@ -33,17 +36,49 @@ import {
 export class LibraryService {
   readonly logger = new Logger(LibraryService.name);
   private access: AccessCore;
+  private configCore: SystemConfigCore;
 
   constructor(
     @Inject(IAccessRepository) accessRepository: IAccessRepository,
     @Inject(IAssetRepository) private assetRepository: IAssetRepository,
+    @Inject(ISystemConfigRepository) configRepository: ISystemConfigRepository,
     @Inject(ICryptoRepository) private cryptoRepository: ICryptoRepository,
     @Inject(IJobRepository) private jobRepository: IJobRepository,
     @Inject(ILibraryRepository) private repository: ILibraryRepository,
     @Inject(IStorageRepository) private storageRepository: IStorageRepository,
     @Inject(IUserRepository) private userRepository: IUserRepository,
   ) {
     this.access = AccessCore.create(accessRepository);
+    this.configCore = SystemConfigCore.create(configRepository);
+  }
+
+  async init() {
+    const config = await this.configCore.getConfig();
+    const libraryScanJob = new CronJob(
+      config.library.scan.cronExpression,
+      async () => {
+        await this.jobRepository.queue({ name: JobName.LIBRARY_QUEUE_SCAN_ALL, data: { force: false } });
+      },
+      // function to run onComplete
+      undefined,
+      // whether it should start directly
+      config.library.scan.enabled,
+      // timezone
+      undefined,
+      // context
+      undefined,
+      // runOnInit
+      undefined,
+      // utcOffset
+      undefined,
+      // prevents memory leaking by automatically stopping when the node process finishes
+      true,
+    );
+
+    this.configCore.config$.subscribe((config) => {
+      libraryScanJob.setTime(new CronTime(config.library.scan.cronExpression));
```

Comment:
nestjs has a default scheduler module, which is exactly for stuff like this. I think we should possibly add new methods to the job repository for job scheduling. So something like
```typescript
  addCronJob(name: string, expression: string);
  updateCronJob(name: string, expression: string);
  removeCronJob(name: string);
```

This way you can write unit tests against it and you don't care about what library or implementation is used behind the scenes. We should probably import the SchedulerRegistry in the infra/job.repository implementation though.

```typescript
constructor(private schedulerRegistry: SchedulerRegistry) {}
```

See https://docs.nestjs.com/techniques/task-scheduling#dynamic-cron-jobs